packages/react/src/components/Storage/FileUploader/hooks/useFileUploader/useFileUploader.tsx

```diff
@@ -1,19 +1,73 @@
+import { checkMaxSize, setAcceptedFiles } from '@aws-amplify/ui';
 import React, { useState } from 'react';
-import { Files } from '../../types';
+import { Files, FileStatuses } from '../../types';
 import { UseFileUploader } from './types';
 
-export default function useFileUploader(): UseFileUploader {
+export default function useFileUploader(
+  maxSize: number,
+  acceptedFileTypes: string[],
+  multiple: boolean
+): UseFileUploader {
+  const [fileStatuses, setFileStatuses] = useState<FileStatuses>([]);
   const [showPreviewer, setShowPreviewer] = useState(false);
   const [files, setFiles] = useState<Files>([]);
 
   const [inDropZone, setInDropZone] = useState(false);
 
-  const addTargetFiles = (targetFiles: FileList) => {
+  const setFileSizeErrors = (files: Files, fileStatuses: FileStatuses) => {
+    const statuses = [...fileStatuses];
+    [...files].forEach((file, index) => {
+      const errorFile = checkMaxSize(maxSize, file);
```

Comment:
The `errorFile` is the message that is returned from the `checkMaxSize`. If it's null, then there was no error, otherwise a string is returned. I wasn't sure if maybe this could be broken up into 2 different calls. One a function that returns a boolean, and another that returns a message, maybe in a static array. Let me tackle that in the next PR.